aexpect/remote.py

```diff
@@ -0,0 +1,1297 @@
+"""
+from __future__ import division
+import logging
+import time
+import re
+import os
+import pipes
+import shutil
+import tempfile
+
+import aexpect
+from avocado.core import exceptions
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest import utils_misc
+from virttest import rss_client
+from virttest.remote_commander import remote_master
+from virttest.remote_commander import messenger
+from virttest.compat_52lts import results_stdout_52lts, results_stderr_52lts
+
+
+class LoginError(Exception):
+
+    def __init__(self, msg, output):
+        Exception.__init__(self, msg, output)
+        self.msg = msg
+        self.output = output
+
+    def __str__(self):
+        return "%s    (output: %r)" % (self.msg, self.output)
+
+
+class LoginAuthenticationError(LoginError):
+    pass
+
+
+class LoginTimeoutError(LoginError):
+
+    def __init__(self, output):
+        LoginError.__init__(self, "Login timeout expired", output)
+
+
+class LoginProcessTerminatedError(LoginError):
+
+    def __init__(self, status, output):
+        LoginError.__init__(self, None, output)
+        self.status = status
+
+    def __str__(self):
+        return ("Client process terminated    (status: %s,    output: %r)" %
+                (self.status, self.output))
+
+
+class LoginBadClientError(LoginError):
+
+    def __init__(self, client):
+        LoginError.__init__(self, None, None)
+        self.client = client
+
+    def __str__(self):
+        return "Unknown remote shell client: %r" % self.client
+
+
+class SCPError(Exception):
+
+    def __init__(self, msg, output):
+        Exception.__init__(self, msg, output)
+        self.msg = msg
+        self.output = output
+
+    def __str__(self):
+        return "%s    (output: %r)" % (self.msg, self.output)
+
+
+class SCPAuthenticationError(SCPError):
+    pass
+
+
+class SCPAuthenticationTimeoutError(SCPAuthenticationError):
+
+    def __init__(self, output):
+        SCPAuthenticationError.__init__(self, "Authentication timeout expired",
+                                        output)
+
+
+class SCPTransferTimeoutError(SCPError):
+
+    def __init__(self, output):
+        SCPError.__init__(self, "Transfer timeout expired", output)
+
+
+class SCPTransferFailedError(SCPError):
+
+    def __init__(self, status, output):
+        SCPError.__init__(self, None, output)
+        self.status = status
+
+    def __str__(self):
+        return ("SCP transfer failed    (status: %s,    output: %r)" %
+                (self.status, self.output))
+
+
+def quote_path(path):
+    """
+    Produce shell escaped version of string item or of list items, which are
+    then joined by space.
+
+    :param path: List or string
+    :return: Shell escaped version
+    """
+    if isinstance(path, list):
+        return ' '.join(map(pipes.quote, path))
+    else:
+        return pipes.quote(path)
+
+
+def handle_prompts(session, username, password, prompt, timeout=10,
```

Comment:
It'd be nice to (later) add default for prompt, what do you think?

Reply:
... and another idea, how about introducing PROMPT_LINUX and PROMPT_WINDOWS global variables as hints?

Reply:
I agree, also the second comment is a really good idea. Both are implemented in the next push.